### DISCUSSION TITLE
ms-python python download failed : Extension with id ms-python.vscode-pylance not found

### DISCUSSION THREAD

**Discussion by Lnncoco** [eclipse-theia/theia#12417] [Q&A]
I want to debug Python files in Theia, so I found the Python plugin.

The version of Theia is 1.36.0.

I added the plugin information in package.json:

```
  "theiaPlugins": {
    "ms-python.python":"https://open-vsx.org/api/ms-python/python/2023.6.0/file/ms-python.python-2023.6.0.vsix"
  }
```

After running the "download:plugins" command, an error was output:

```
+ ms-python.python: downloaded successfully 
--- collecting extension-packs ---
--- resolving 2 extension-packs ---
+ ms-toolsai.jupyter@2022.10.110: downloaded successfully 
--- collecting extension dependencies ---
Extension with id ms-python.vscode-pylance not found at https://open-vsx.org/api/-/query?extensionId=ms-python.vscode-pylance&includeAllVersions=true
Error: Errors downloading some plugins. To make these errors non fatal, re-run with --ignore-errors
    at downloadPlugins (/home/theia/node_modules/@theia/cli/lib/download-plugins.js:116:15)
    at async Object.handler (/home/theia/node_modules/@theia/cli/lib/theia.js:314:13)
error Command failed with exit code 1.
```

I searched for discussions and it seems that ms-python. vscode-pylance has been closed source, which makes it impossible to install normally? 

Is there any way to debug Python in Theia?

**Comment by msujew** [comment]
Hey @Lnncoco,

I've provided an answer on why Pylance isn't installable (licensing issues) and what to use instead in [this issue](https://github.com/eclipse-theia/theia/issues/12219#issuecomment-1443535930).
